### PR TITLE
Molecule idrac_gather_facts enhancement

### PR DESCRIPTION
**Pull request by Kritika-Bhateja-03**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: roles/idrac_gather_facts/molecule/systemmetrics/converge.yml [hunk 1]
```diff
@@ -98,4 +83,4 @@
       loop_control:
         loop_var: power_data
         index_var: index
-      when: power_metrics | length > 0
+      when: power_metrics | length > 0 and and psu_ids | length > 0
```
Comment (rajshekarp87) [inline roles/idrac_gather_facts/molecule/systemmetrics/converge.yml]:
```suggestion
      when: power_metrics | length > 0 and psu_ids | length > 0
```

Comment (Kritika-Bhateja-03) [reply roles/idrac_gather_facts/molecule/systemmetrics/converge.yml]: done

**Comment by Kritika-Bhateja-03** [comment]
[idrac_gather_facts_16g.zip](https://github.com/dell/dellemc-openmanage-ansible-modules/files/15153959/idrac_gather_facts_16g.zip)

**Comment by Kritika-Bhateja-03** [comment]
[idrac_gather_facts_15g.zip](https://github.com/dell/dellemc-openmanage-ansible-modules/files/15158652/idrac_gather_facts_15g.zip)
[idrac_gather_facts_6g_server2.zip](https://github.com/dell/dellemc-openmanage-ansible-modules/files/15158653/idrac_gather_facts_6g_server2.zip)

**Review by rajshekarp87 — APPROVED** [review]

**Review by jagadeeshnv — APPROVED** [review]

**Review by singh-lovepreet1 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**